File: modulesCore/loginAndRegister/src/main/java/cn/pivotstudio/modulec/loginandregister/repository/SetPasswordRepository.java
package cn.pivotstudio.modulec.loginandregister.repository;

import android.annotation.SuppressLint;

import androidx.lifecycle.MutableLiveData;

import java.util.HashMap;

import cn.pivotstudio.husthole.moduleb.network.BaseObserver;
import cn.pivotstudio.husthole.moduleb.network.NetworkApi;

import cn.pivotstudio.husthole.moduleb.network.errorhandler.ExceptionHandler;

import cn.pivotstudio.modulec.loginandregister.model.MsgResponse;

import cn.pivotstudio.modulec.loginandregister.network.LRRequestInterface;

/**
 * @classname:SetPasswordRepository
 * @description:
 * @date:2022/5/1 21:33
 * @version:1.0
 * @author:
 */
@SuppressLint("CheckResult")
public class SetPasswordRepository {
    public final MutableLiveData<String> failed = new MutableLiveData<>();
    public MutableLiveData<MsgResponse> mSetPassword = new MutableLiveData<>();
    public SetPasswordRepository() {
    }
    public void createAccountForNetwork(String id, String password) {
        HashMap<String, String> map = new HashMap<>();
        map.put("email", id);
        map.put("password",password);
        NetworkApi.createService(LRRequestInterface.class, 2).
                register(map).compose(NetworkApi.applySchedulers(new BaseObserver<MsgResponse>() {
            @Override
            public void onSuccess(MsgResponse setPasswordResponse) {
                mSetPassword.setValue(setPasswordResponse);

            }

            @Override
            public void onFailure(Throwable e) {
                failed.postValue(((ExceptionHandler.ResponseThrowable) e).message);

            }
        }));
    }
}

File: modulesCore/loginAndRegister/src/main/java/cn/pivotstudio/modulec/loginandregister/repository/SetPasswordRepository.kt
package cn.pivotstudio.modulec.loginandregister.repository

import android.annotation.SuppressLint
import androidx.lifecycle.MutableLiveData
import cn.pivotstudio.husthole.moduleb.network.NetworkApi
import cn.pivotstudio.modulec.loginandregister.network.LRRequestInterface
import cn.pivotstudio.husthole.moduleb.network.BaseObserver
import cn.pivotstudio.husthole.moduleb.network.errorhandler.ExceptionHandler.ResponseThrowable
import cn.pivotstudio.modulec.loginandregister.model.MsgResponse
import cn.pivotstudio.modulec.loginandregister.network.LoginAndRegisterNetworkApi
import java.util.HashMap

/**
 * @classname:SetPasswordRepository
 * @description:
 * @date:2022/5/1 21:33
 * @version:1.0
 * @author:
 */
@SuppressLint("CheckResult")
class SetPasswordRepository {
    @JvmField
    val failed = MutableLiveData<String>()

    @JvmField
    var mSetPassword = MutableLiveData<MsgResponse>()

    fun createAccountForNetwork(id: String, password: String) {
        val map = HashMap<String, String>()
        map["email"] = id
        map["password"] = password
        LoginAndRegisterNetworkApi.retrofitService
            .register(map)
            .compose(NetworkApi.applySchedulers(object : BaseObserver<MsgResponse>() {
                override fun onSuccess(setPasswordResponse: MsgResponse) {
                    mSetPassword.value = setPasswordResponse
                }

                override fun onFailure(e: Throwable) {
                    failed.postValue((e as ResponseThrowable).message)
                }
            }))
    }
}